pkg/cli/scan.go

```diff
@@ -615,13 +628,14 @@ func resolveInputFileFromArg(inputFilePath string) (*os.File, error) {
 // For example, given the input value "calico", this function will find the
 // latest version of the package (e.g. "calico-3.26.3-r3.apk") and download it
 // for each architecture.
-func resolveInputForRemoteTarget(ctx context.Context, input string) (downloadedAPKFilePaths []string, cleanup func() error, err error) {
+func resolveInputForRemoteTarget(ctx context.Context, input, repository string) (downloadedAPKFilePaths []string, cleanup func() error, err error) {
 	logger := clog.FromContext(ctx)
 
 	archesFound := 0
 	for _, arch := range []string{"x86_64", "aarch64"} {
-		const apkRepositoryURL = "https://packages.wolfi.dev/os"
-		apkindex, err := index.Index(arch, apkRepositoryURL)
+		// Since index.Index function doesn't respect the `$HTTP_AUTH`, use
+		// fetchAPKIndex function instead.
+		apkindex, _, err := fetchAPKIndex(ctx, getRepositoryURL(repository, arch))
```

Comment:
`fetchAPKIndex` looks like it's much more complex than we need (e.g. we don't want to be handling stdin or local index files or anything here, and it's meant to support a separate CLI command).

I see we used `addAuth` to get auth for an HTTP request below — can we not use that here for some reason?

Reply:
We already using `addAuth` below but we also need to use `fetchAPKIndex` since it does `addAuth`. 🤷

Reply:
We can clean this up after the PR merges. Ideally we've have a less complicated way to fetch APK indexes, but I think this is a more global issue in our code than just right here 👍 cc: @imjasonh